wgengine/userspace.go

```diff
@@ -461,6 +493,17 @@ func (e *userspaceEngine) Reconfig(cfg *wgcfg.Config, routerCfg *router.Config)
 		panic("routerCfg must not be nil")
 	}
 
+	localAddrs := map[packet.IP]bool{}
+	for _, addr := range routerCfg.LocalAddrs {
+		// TODO: ipv6
+		if !addr.IP.Is4() {
+			continue
+		}
+		bs := addr.IP.As16()
+		localAddrs[packet.NewIP(net.IP(bs[12:16]))] = true
```

Comment:
the `12:16` is avoidable. Just use addr.IP.To4? (can even do it earlier and do the `continue` if it's nil)

Reply:
`addr.IP` is a netaddr.IP, which only has As16, not To4.

Really this is another wart that's asking to replace packet.IP with netaddr.IP throughout, so I didn't push too hard on making it pretty. The real fix is to stop using packet.IP.